app/src/app/admin/articles/listing/listing.ts

```diff
@@ -45,23 +51,86 @@ namespace app.admin.articles.listing {
 
     }
 
+    /* istanbul ignore next:@todo - skipping controller testing */
     export class ArticlesListingController {
 
-        public articles:common.models.Article[] = [];
-        static $inject = ['articlesPaginator', 'initArticles', '$stateParams'];
+        static $inject = ['articlesPaginator', 'tagsPaginator', 'usersPaginator', 'initArticles', '$stateParams'];
 
+        public articles:common.models.Article[] = [];
         public pages:number[] = [];
-
         public currentPageIndex:number;
-
-        constructor(private articlesPaginator:common.services.pagination.Paginator, articles, public $stateParams:IArticlesListingStateParams) {
+        public tagsToFilter:common.models.Tag[] = [];
+        public usersToFilter:common.models.User[] = [];
+        public queryString:string;
+
+        constructor(
+            private articlesPaginator:common.services.pagination.Paginator,
+            private tagsPaginator:common.services.pagination.Paginator,
+            private usersPaginator:common.services.pagination.Paginator,
+            articles,
+            public $stateParams:IArticlesListingStateParams
+        ) {
 
             this.articles = articles;
 
             this.pages = articlesPaginator.getPages();
 
             this.currentPageIndex = this.$stateParams.page - 1;
+
         }
+
+        /**
+         * Function called when article is searched for.
+         */
+        public search():void {
```

Comment:
In general it is better to return the promise that this function calls as it is easier to unit test.